lib/puppet/type/cisco_stp_global.rb

```diff
@@ -0,0 +1,755 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+begin
+  require 'puppet_x/cisco/cmnutils'
+rescue LoadError # seen on master, not on agent
+  # See longstanding Puppet issues #4248, #7316, #14073, #14149, etc. Ugh.
+  require File.expand_path(File.join(File.dirname(__FILE__), '..', '..',
+                                     'puppet_x', 'cisco', 'cmnutils.rb'))
+end
+
+Puppet::Type.newtype(:cisco_stp_global) do
+  @doc = "
+    Manages the Cisco Spanning-tree Global configuration resource.
+    cisco_stp_global {'default':
+      ..attributes..
+    }
+    'default' is only acceptable name for this global config object.
+    Example:
+    cisco_portchannel_global { 'default':
+      bd_designated_priority       => [['2-42', '40960'], ['83-92,1000-2300', '53248']],
+      bd_forward_time              => [['2-42', '26'], ['83-92,1000-2300', '20']],
+      bd_hello_time                => [['2-42', '6'], ['83-92,1000-2300', '9']],
+      bd_max_age                   => [['2-42', '26'], ['83-92,1000-2300', '20']],
+      bd_priority                  => [['2-42', '40960'], ['83-92,1000-2300', '53248']],
+      bd_root_priority             => [['2-42', '40960'], ['83-92,1000-2300', '53248']],
+      bpdufilter                   => true,
+      bpduguard                    => true,
+      bridge_assurance             => false,
+      domain                       => 100,
+      fcoe                         => false,
+      loopguard                    => true,
+      mode                         => 'mst',
+      mst_designated_priority      => [['2-42', '40960'], ['83-92,1000-2300', '53248']],
+      mst_forward_time             => 25,
+      mst_hello_time               => 5,
+      mst_inst_vlan_map            => [['2', '6-47'], ['92', '120-400']],
+      mst_max_age                  => 35,
+      mst_max_hops                 => 200,
+      mst_name                     => 'nexus',
+      mst_priority                 => [['2-42', '40960'], ['83-92,1000-2300', '53248']],
+      mst_revision                 => 34,
+      mst_root_priority            => [['2-42', '40960'], ['83-92,1000-2300', '53248']],
+      pathcost                     => 'long',
+      vlan_designated_priority     => [['1-42', '40960'], ['83-92,1000-2300', '53248']],
+      vlan_forward_time            => [['1-42', '19'], ['83-92,1000-2300', '13']],
+      vlan_hello_time              => [['1-42', '10'], ['83-92,1000-2300', '6']],
+      vlan_max_age                 => [['1-42', '21'], ['83-92,1000-2300', '13']],
+      vlan_priority                => [['1-42', '40960'], ['83-92,1000-2300', '53248']],
+      vlan_root_priority           => [['1-42', '40960'], ['83-92,1000-2300', '53248']],
+    }
```

Comment:
Added some information above the Example: section, let me know if this is ok or anything else is needed.